src/Compilers/CSharp/Portable/Symbols/Attributes/RetargetingAttributeData.cs

```diff
@@ -4,43 +4,52 @@
+            CSharpAttributeData underlying,
             NamedTypeSymbol attributeClass,
-            MethodSymbol attributeConstructor,
+            MethodSymbol? attributeConstructor,
             ImmutableArray<TypedConstant> constructorArguments,
-            ImmutableArray<int> constructorArgumentsSourceIndices,
-            ImmutableArray<KeyValuePair<string, TypedConstant>> namedArguments,
-            bool hasErrors,
-            bool isConditionallyOmitted)
-            : base(applicationNode, attributeClass, attributeConstructor, constructorArguments, constructorArgumentsSourceIndices, namedArguments, hasErrors, isConditionallyOmitted)
+            ImmutableArray<KeyValuePair<string, TypedConstant>> namedArguments)
         {
+            Debug.Assert(underlying is SourceAttributeData or SynthesizedAttributeData);
+
+            _underlying = underlying;
+            _attributeClass = attributeClass;
+            _attributeConstructor = attributeConstructor;
+            _constructorArguments = constructorArguments;
+            _namedArguments = namedArguments;
         }
 
-        /// <summary>
-        /// Gets the retargeted System.Type type symbol.
-        /// </summary>
-        /// <param name="targetSymbol">Target symbol on which this attribute is applied.</param>
-        /// <returns>Retargeted System.Type type symbol.</returns>
-        internal override TypeSymbol GetSystemType(Symbol targetSymbol)
-        {
-            var retargetingAssembly = (RetargetingAssemblySymbol)(targetSymbol.Kind == SymbolKind.Assembly ? targetSymbol : targetSymbol.ContainingAssembly);
-            var underlyingAssembly = (SourceAssemblySymbol)retargetingAssembly.UnderlyingAssembly;
+        public override NamedTypeSymbol AttributeClass => _attributeClass;
+        public override MethodSymbol? AttributeConstructor => _attributeConstructor;
+        protected internal override ImmutableArray<TypedConstant> CommonConstructorArguments => _constructorArguments;
+        protected internal override ImmutableArray<KeyValuePair<string, TypedConstant>> CommonNamedArguments => _namedArguments;
 
-            // Get the System.Type from the underlying assembly's Compilation
-            TypeSymbol systemType = underlyingAssembly.DeclaringCompilation.GetWellKnownType(WellKnownType.System_Type);
+        public override SyntaxReference? ApplicationSyntaxReference => null;
 
-            // Retarget the type
-            var retargetingModule = (RetargetingModuleSymbol)retargetingAssembly.Modules[0];
-            return retargetingModule.RetargetingTranslator.Retarget(systemType, RetargetOptions.RetargetPrimitiveTypesByTypeCode);
-        }
+        [MemberNotNullWhen(true, nameof(AttributeClass), nameof(AttributeConstructor))]
+        internal override bool HasErrors => _underlying.HasErrors || _attributeConstructor is null;
+
+        internal override bool IsConditionallyOmitted => _underlying.IsConditionallyOmitted;
```

Comment:
Not related to this PR: I understand this preserves existing behavior, but it seems a bit strange. The xml doc for `CommonAttributeData.IsConditionallyOmitted` says that the value should only be true for source attributes.

Reply:
For the most part, RetargetingAttributeData represents an attribute in source